### ISSUE TITLE
Feat:Train Yolov8n model

### ISSUE THREAD

**Issue by Rishab87** [AOSSIE-Org/PictoPy#246]
### Describe the feature

To classify images in more categories we need to train yolov8n model as mentioned in PR #245, I tried training this model with model.train method of ultralytics but was not able to find a suitable dataset with bounding labels. So need to find a suitable dataset and train it on that to add more classes

### Add ScreenShots

No screenshots to attatch

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

**Comment by Rishab87** [comment]
If I find any dataset I'll train and create PR for it, meanwhile if anyone finds dataset and want to collaborate on this do tag me!

**Comment by kamisama-coder** [comment]
you only want dataset or the help in setting model

**Comment by Rishab87** [comment]
I only want the dataset

**Comment by Priyansh-max** [comment]
The yolo model is pre trained on 80 classes u need more than that?

**Comment by Rishab87** [comment]
yes to categorise images on more than those classes like screenshots, nature etc

**Comment by IWaSReLoAdInG08** [comment]
@Rishab87  I would like to work on this issue and contribute to expanding the classification capabilities of the YOLOv8n model.

As mentioned in the issue description, finding a suitable dataset with bounding box labels is crucial for training the model to classify images into more categories. I currently have access to a traffic dataset that includes annotations for various classes such as:

Vehicles (cars, trucks, buses, motorcycles, etc.)
Pedestrians
Other relevant objects (e.g., traffic lights, road signs, bicycles, etc.)
This dataset seems well-suited for training the YOLOv8n model to detect and classify objects in traffic scenarios. It also aligns with the goal of adding more classes to the model.

**Comment by Rishab87** [comment]
The classes which you mentioned here, this model is already trained on those classes, were you able to find dataset with other classes than these? (for eg screenshots, plants etc). If yes please do tell

**Comment by Rishab87** [comment]
Current classes of model:
'''python
class_names = ['person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
               'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
               'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
               'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
               'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
               'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
               'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard',
               'cell phone', 'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase',
               'scissors', 'teddy bear', 'hair drier', 'toothbrush', 'unknown']
'''

**Comment by ShivanshuVerma8** [comment]
There are datasets which include more than 20000 categories like imagenet and big detection.  In the dataset I mentioned above I can make a simulation based one and include categories like plants and all .

**Comment by Rishab87** [comment]
Just to confirm do they contain bounding labels?

**Comment by IWaSReLoAdInG08** [comment]
Yes they have labels too. I have spent considerable time on this and gone through research papers and found some notable datasets with labels and bounding boxes and other image annotations that classify objects .

**Comment by Rishab87** [comment]
If you want to train with these datasets then you can go ahead otherwise you can link those datasets here, I can train them I've evrything setup for it.

**Comment by IWaSReLoAdInG08** [comment]
I will train them too , and also i will link the datasets here :
1. [OpenImage](https://storage.googleapis.com/openimages/web/index.html?spm=5aebb161.5e6c11a6.0.0.1442c921jjFpz5)
2. Subset of this : https://github.com/EscVM/OIDv4_ToolKit?spm=5aebb161.5e6c11a6.0.0.1442c921jjFpz5
Conversion Tool :  Roboflow to filter and convert annotations.

2. [VisDrone](https://github.com/VisDrone/VisDrone-Dataset?tab=readme-ov-file)
Already in YOLO format . No need for conversion

There is more please confirm if you need something like this .

**Comment by Rishab87** [comment]
I think this would work!

**Comment by IWaSReLoAdInG08** [comment]
@Rishab87 I have already written a script to fetch the model and have implemented some initial code as a starting point. Once the training process is complete, I would be more than happy to collaborate further and contribute meaningfully to the project. I look forward to working together and helping building this feature.

**Comment by Rishab87** [comment]
@ShivanshuVerma83 Are you done with this or do you need any help or should I train it?

**Comment by ShivanshuVerma8** [comment]
I have trained and wrote the script but due to limited GPU restrictions, I haven't done it for more number of classes , you can go ahead with the training . I will try to do for more number of classes, and I asked for help in discord no one responded there on how to go ahead with few issues. I resolved it though and also I raise a PR regarding some minor issues please have a look if you can.

**Comment by Rishab87** [comment]
Sorry I was busy with the exams was not active on discord, can you provide me the script and dataset which you are using for it to get an idea what you were doing, I'll train this model then further, also I saw your PR but I'm not the maintainer so I can't merge let the maintainer come back and see it

**Comment by ShivanshuVerma8** [comment]
No worries, I will share with you the script by Tuesday,  as I am out and dont have my pc with me . I was thinking to use GPUs available on kaggle. Can you tell me any other way if you have !

**Comment by IWaSReLoAdInG08** [comment]
https://github.com/ShivanshuVerma83/train-yolov8-custom-dataset-step-by-step-guide
Hi @Rishab87 Have a look i followed this approach , i dont know if it is right or wrong . Have a look and can you share the script you guys used to train your model .

**Comment by Rishab87** [comment]
@ShivanshuVerma8 thanks for sharing this, script was something like this:
```python
def train_yolo():
    from ultralytics import YOLO
    
    model = YOLO('yolov8n.pt')
    
    results = model.train(
        data='dataset.yaml',
        epochs=100,
        imgsz=640,
        batch=16,
        name='tiny-imagenet'
    )
    
    # Export to ONNX
    model.export(format='onnx')

```
